std/stdio.d

```diff
@@ -4091,14 +4091,84 @@ unittest
     }
 }
 
+// roll our own appender, but with "safe" arrays
+private struct ReadlnAppender
+{
+    import core.stdc.string;
+
+    char[] buf;
+    size_t cap;
+    size_t pos;
```

Comment:
The original length of the buffer is needed in `data()` to detect whether the buffer was reallocated.